patchutils.go

```diff
@@ -594,6 +711,24 @@ func convertChunksIntoFileDiff(chunks []dbd.Chunk, fileDiff *diff.FileDiff) {
 	fileDiff.Hunks = append(fileDiff.Hunks, currentHunk)
 }
 
+// interPrintSingleFileDiff returns printed version of diffFile, which was found only in one out of two versions.
+// NOTE: hunks of diffFile might be reverted after calling of this function.
```

Comment:
You may want to avoid that by passing diffFile by value instead of passing it by reference. While the caveat is documented, it is easy to miss and thus likely to lead to bugs, so better avoid it completely.

Reply:
Oh, actually that `NOTE` isn't relevant right now. I removed reverting hunks if needed, out of this function. For example: https://github.com/google/go-patchutils/blob/25854ddbcb317fb505cdfaab37d17e07834c9c43/patchutils.go#L80

So, right now this function doesn't change anything in passed diff.FileDiff. Leaving it as a reference.